### ISSUE TITLE
SOCKS5 support

### ISSUE THREAD

**Issue by t-900-a** [openmls/openmls#303]
**Is your feature request related to a problem? Please describe.**
Adding SOCKS5 support would open up the possibility of routing traffic from client to server via i2p, nym, tor, etc

**Describe the solution you'd like**
SOCKS5 support

**Describe alternatives you've considered**
Routing network traffic at the OS level, rather than the application level

**Additional context**

**Comment by raphaelrobert** [comment]
Hi!
I suppose that's related to the cli PoC? The library itself doesn't do any networking.

**Comment by t-900-a** [comment]
yeah it's too early for socks5 support, just something to keep in mind.
